### PR TITLE
Fix for removing team from mutable object in cached map

### PR DESCRIPTION
**Pull request by ThetaDR**
# Contribution checklist

## Brief description
https://front.hc.engineering/workbench/platform/tracker/TSK-667

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [x] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [x] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

**Comment by ThetaDR** [comment]
Another option to prevent stuff like this in future: store immutable object in queryCache https://github.com/hcengineering/anticrm/blob/fa9259ad18b0b25576b3703c7224ac852e2c892d/server/core/src/utils.ts#L25-L31
@haiodo, @BykhovDenis what do you think?

**Comment by BykhovDenis** [comment]
> Another option to prevent stuff like this in future: store immutable object in queryCache
> 
> https://github.com/hcengineering/anticrm/blob/fa9259ad18b0b25576b3703c7224ac852e2c892d/server/core/src/utils.ts#L25-L31
> 
> 
> @haiodo, @BykhovDenis what do you think?

I think it's better to return a clone of the result from the cache

File: server-plugins/tracker-resources/src/index.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ async function updateSubIssues (
  */
 export async function issueHTMLPresenter (doc: Doc, control: TriggerControl): Promise<string> {
   const issue = doc as Issue
-  const team = (await control.findAll(tracker.class.Team, { _id: issue.space })).shift()
+  const team = (await control.findAll(tracker.class.Team, { _id: issue.space }))[0]
```
Comment (haiodo) [inline server-plugins/tracker-resources/src/index.ts]: Better to update cache and return copied values. Please use Hierarchy.clone()

File: server/core/src/utils.ts [hunk 2]
```diff
@@ -15,6 +16,7 @@ import {
 export function createCacheFindAll (storage: ServerStorage): ServerStorage['findAll'] {
   // We will cache all queries for same objects for all derived data checks.
   const queryCache = new Map<string, FindResult<Doc>>()
+  const hierarchy = new Hierarchy()
```
Comment (BykhovDenis) [inline server/core/src/utils.ts]: It will not work, you have hierarchy in `storage.hierarchy`

**Review by BykhovDenis — CHANGES_REQUESTED** [review]

File: server/core/src/utils.ts (hunk 2 above)
Comment (ThetaDR) [reply server/core/src/utils.ts]: Fixed, thank you

**Review by BykhovDenis — APPROVED** [review]

**Review by haiodo — APPROVED** [review]

**Pull request merged**